Add what's new in WinForms article (#114)
dotnet/docs-desktop: dotnet-desktop-guide/net/winforms/whats-new/index.md

---
title: What's new in Windows Forms
description: Learn about what's new in Windows Forms for .NET. Windows Forms. .NET provides new features and enhancements over .NET Framework.
ms.date: 11/05/2020
ms.topic: conceptual
---

# What's new (Windows Forms.NET)

> Windows Forms for .NET adds the following features and enhancements over .NET Framework.

Reviewer: From .NET Fx to .NET sounds awkward, might be better to specify .NET 5 at least once or twice up front.

```suggestion
Windows Forms for .NET 5.0 adds the following features and enhancements over .NET Framework.
```

> There are a few breaking changes you should be aware of when migrating from .NET Framework to .NET. For more information, see [Breaking changes in Windows Forms](/dotnet/core/compatibility/winforms).

Reviewer: ```suggestion
There are a few breaking changes you should be aware of when migrating from .NET Framework to .NET 5.0. For more information, see [Breaking changes in Windows Forms](/dotnet/core/compatibility/winforms).
```

## Enhanced features

- Microsoft UI Automation patterns work better with accessibility tools like Narrator and Jaws.

> - Improved performance and reliability.

Reviewer 2: As per the email, "reliability" is probably a misplaced term here. Windows Forms on .NET Framework was quite reliable.

> - Significant work on performance and reliability.

Reviewer: ```suggestion
- Improved performance and reliability.
```

> - Better support for high DPI resolutions such as 4k monitors.

Reviewer 2: We haven't done much for "better support", so I'd be cautious mentioning this. 
We have enhanced the default VB app template in be SystemAware, to match we've done for C# in .NET Core 3.0. The better high DPI support is currently slated for .NET 6.0.

- The default font matches the current Windows design recommendations.

  > [!CAUTION]
  > This may impact the layout of apps migrated from .NET Framework.

## New controls

> The following new controls have been added since .NET Framework:

Reviewer: ```suggestion
The following controls have been added since Windows Forms was ported to .NET Framework:
```

- <xref:System.Windows.Forms.TaskDialog?displayProperty=fullName>
  
  A task dialog is a dialog box that can be used to display information and receive simple input from the user. Like a message box, it's formatted by the operating system according to parameters you set. Task dialog has more features than a message box. For more information, see the [Task dialog sample](https://github.com/dotnet/samples/tree/master/windowsforms/TaskDialogDemo).

- <xref:Microsoft.Web.WebView2.WinForms.WebView2?displayProperty=fullName>

  A new web browser control with modern web support. Based on Edge (Chromium). For more information, see [Getting started with WebView2 in Windows Forms](/microsoft-edge/webview2/gettingstarted/winforms).

## Enhanced controls

- <xref:System.Windows.Forms.ListView?displayProperty=fullName>

  - Supports collapsible groups.
  - Footers

>   - Group subtitle, task, title images, and collapse/expand functionality.

Reviewer 2: Does each list item need to end with a "."? If so, the second line needs a ".".

>   - Group subtitle, task, title images, and collapse/expand functionality.

Reviewer 2: The last point is a dup of the first point, isn't it?
